src/main/java/bio/terra/service/dataset/flight/ingest/DatasetIngestFlight.java

```diff
@@ -218,6 +218,7 @@ public DatasetIngestFlight(FlightMap inputParameters, Object applicationContext)
         // Ensure that no duplicate IDs are being loaded in
         addStep(new IngestValidateIngestRowsStep(datasetService));
         if (mergeIngest) {
+          addStep(new IngestValidatePrimaryKeyDefinedStep(datasetService));
```

Comment:
I think it makes sense to do this check for the `replace` ingest strategy as well

Reply:
Actually, in `replace` ingests we have a fallback behavior in the [specification doc](https://docs.google.com/document/d/1MhWxY6lHYOvVKmUlLhJGNJ8M79gYycc0cUdvbBE9n_Q/edit):

>If no primary key is present in the table being ingested to, it will fall back to append mode.

Reply:
Oh interesting...I wonder if we still want to do that 🤔 If we append the records, we would have duplicate values for the primary keys right? I wonder if users would be caught off-guard by that

Reply:
I do think this is still what we want, primary key uniqueness is not enforced for other types of ingest for (I believe) historical reasons but @nmalfroy may be able to share more / we can chat in stand-up!  I agree that it's not quite what one may expect.

Either way, I'd designate any changes to `replace` ingest behavior as being OOS / to be ticketed separately.

Reply:
Yeah, I was kinda torn originally.  I think if we change the behavior we should alert users but yeah, I suppose it might make sense to fail.  Also agree it's out of scope for this particular token